lib/Support/CustomDirectiveImpl.cpp

```diff
@@ -34,6 +52,21 @@ bool circt::inferImplicitSSAName(OpAsmParser &parser, NamedAttrList &attrs) {
   return true;
 }
 
+void circt::printImplicitSSAName(OpAsmPrinter &printer, Operation *op,
+                                 StringAttr attr) {
+  SmallString<32> resultNameStr;
+  llvm::raw_svector_ostream tmpStream(resultNameStr);
+  printer.printOperand(op->getResult(0), tmpStream);
+  auto actualName = tmpStream.str().drop_front();
+  auto expectedName = attr.getValue();
+  // Anonymous names are printed as digits, which is fine.
+  if (actualName == expectedName ||
+      (expectedName.empty() && isdigit(actualName[0])))
+    return;
+
+  printer << "name " << attr;
```

Comment:
Since the SystemC operations also support this `name` attribute now, we might want to add a test to check the assembly format for `systemc.instance.decl` and `systemc.signal` for the new case.

Reply:
Done!